packages/react-server/src/ReactFlightServer.js

```diff
@@ -3995,8 +3998,9 @@ function retryTask(request: Request, task: Task): void {
       emitModelChunk(request, task.id, json);
     }
 
-    request.abortableTasks.delete(task);
     task.status = COMPLETED;
+    request.abortableTasks.delete(task);
+    callOnAllReadyIfReady(request);
```

Comment:
We could handle tasks in performWork and listeners in the callbacks. The only time we'd clear a task is if we're in a performWork loop so we can always be certain we'll handle those there.